ml-agents/mlagents/trainers/trainer_util.py

```diff
@@ -211,13 +219,23 @@ def handle_existing_directories(
     if model_path_exists:
         if not resume and not force:
             raise UnityTrainerException(
-                "Previous data from this run-id was found. "
-                "Either specify a new run-id, use --resume to resume this run, "
+                "Previous data from this run id was found. "
+                "Either specify a new run id, use --resume to resume this run, "
                 "or use the --force parameter to overwrite existing data."
             )
     else:
         if resume:
             raise UnityTrainerException(
-                "Previous data from this run-id was not found. "
+                "Previous data from this run id was not found. "
                 "Train a new run by removing the --resume flag."
             )
+
+    # Verify init path if specified.
```

Comment:
Could this live closer to the entrypoint, like in `learn.py` itself?  Could allow us to fail faster.

Reply:
This function is actually called pretty early on in the `learn.py`. I just wanted to put all of this type of functionality in one place as the `learn.py` script is quite long. We can move it out if that's cleaner.

Reply:
Makes sense,  I actually think I misread this at the time as `trainer.py` rather than `trainer_util.py`